rust/lance/src/index/vector/pq.rs

```diff
@@ -408,6 +408,19 @@ pub async fn build_pq_model(
     } else {
         training_data
     };
+
```

Comment:
this logic might be similar with what we have in `_verify_pq_sample_rate`, maybe we can modify `_verify_pq_sample_rate` to have `num_codes`

Reply:
We have the parameter `num_bits` which can be used to control the number of codes, but using fewer bits/codes could negatively affect index recall